src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1587,6 +1682,7 @@ private static TypeWithState GetParameterState(TypeWithAnnotations parameterType
         protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement node)
         {
             Debug.Assert(!IsConditionalState);
+            EnforceMemberNotNull(node.Syntax);
```

Comment:
>EnforceMemberNotNull [](start = 12, length = 20)

This placement of the check is not correct.  It does not account for the fact that the return statement may be inside a try-finally where the member is given the proper state inside the finally.  This needs to be done at the top level by processing the set of pending returns.